### PR TITLE
[Enhancement#60] Use 'Ctrl + Enter' to submit comment

### PR DESCRIPTION
**Pull request by LaChope**
Comment should be submitted by pressing 'Ctrl + Enter'.

### CODE REVIEW

File: src/components/comment/CommentView.jsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ const CommentView = (props) => {
                 <span className="col-auto text-muted comment-timestamp">{timeAgo}</span>
             </div>
             <div className="row">
-                <span className="col comment-value">{props.commentValue}</span>
+                <span className="col comment-value" style={ { whiteSpace: "pre-line" } }>{props.commentValue}</span>
```
Comment (blcham) [inline src/components/comment/CommentView.jsx:17]: Doesn't it make more sense to put it into CSS file ? Why do we have it here ?

**Review by blcham — CHANGES_REQUESTED** [review]
Source code seems all right, see the minor issues.

Thus I started testing and found following issues:
1) comments are not persisted
- first i create 3 comments
Selection_104

- then I move to "Conditional Logic" and back. Now i see that my comments are lost.
Selection_105


2) modal dialog is not responsive 
- if i show the dialog:
Selection_102

- and then make screen smaller, it goes outside of the screen:
Selection_103

**Comment by LaChope** [comment]
@blcham I cannot reproduce your first issue with non persisting comments.

- image

- image

- image

- image

File: src/components/wizard/VerticalWizardNav.jsx [hunk 2]
```diff
@@ -5,36 +5,39 @@ import JsonLdUtils from 'jsonld-utils';
...
+                        key={'nav' + index}
+                        onClick={() => onNavigate(index)}
+                        id={'wizard-nav-' + index}
+                        action={true}
+                        active={index === currentStep ? 'active' : ''}
+                        variant={'default'}
+                        className={Question.getEmphasizedClass(step)}
+                    >
+                        {JsonLdUtils.getLocalized(step[JsonLdUtils.RDFS_LABEL], options.intl)}
+                        {/*<QuestionCommentIcon question={step}/>*/}
```
[the remark this answers is not in the document]
Comment (blcham) [reply src/components/wizard/VerticalWizardNav.jsx]: sure

**Comment by LaChope** [comment]
Ok I could reproduce the bug now, I will try to see if it is possible to lift the state up or using localStorage.

File: src/components/comment/CommentList.jsx [hunk 3]
```diff
@@ -5,20 +5,28 @@ import Constants from "../../constants/Constants";
...
-                        timestamp={comment[Constants.HAS_TIMESTAMP]}
-                    />
-                </div>
-            ))}
-        </>
-    )
+    const addComments = () => {
+        return (
+            props.comment.map((comment) => (
+            <div key={JSON.stringify(comment[Constants.HAS_TIMESTAMP])}
```
Comment (blcham) [inline src/components/comment/CommentList.jsx:11]: Why we need here JSON.stringify ?

File: src/components/comment/CommentView.jsx [hunk 4]
```diff
@@ -1,22 +1,50 @@
...
+        if (props.author) {
+            return parseJsonComment(props.author, [Constants.HAS_AUTHOR]);
+        } else return UNKNOWN_AUTHOR;
+    }
+
+    const getSplitAuthor = () => {
+        let commentAuthor = {};
+        if (props.author) {
+            commentAuthor = parseJsonComment(props.author, [Constants.HAS_AUTHOR]);
+            return "..." + commentAuthor.split('/people/').pop();
```
Comment (blcham) [inline src/components/comment/CommentView.jsx:29]: This is not correct as you are assuming that url will contain string people. The url can be any url you can imagine.

Take only localName i.e. last fragment of url. Fragments are separated by "/" or "#".

File: src/components/comment/CommentView.jsx [hunk 5]
```diff
@@ -1,22 +1,50 @@
...
 
+const UNKNOWN_AUTHOR = "Unknown author";
+
 const CommentView = (props) => {
+    TimeAgo.addLocale(en);
+    const time = new TimeAgo('en-US');
+
+    const [show, setShow] = useState(false);
+
     const parseJsonComment = (value, constant) => {
```
Comment (blcham) [inline src/components/comment/CommentView.jsx:14]: I dont understand what this method and it is very wird.
Why it has generic parameters (value, constant) which does not say anything and then you even pass array:
[Constants.HAS_AUTHOR]

And why it is called parseJsonComment if it does not have anything todo with comment. Moreover, i am quite sure we will have such functionality in some utility class. I would search for something like JsonldUtils ... and if we don't have, it belongs there.

File: src/components/comment/CommentView.jsx [hunk 6]
```diff
@@ -1,22 +1,50 @@
...
         return jsonComment[constant];
     }
 
-    TimeAgo.addLocale(en);
-    const time = new TimeAgo('en-US');
-    const timeAgo = time.format(parseJsonComment(props.timestamp, [Constants.HAS_TIMESTAMP]));
+    const getAuthor = () => {
+        if (props.author) {
+            return parseJsonComment(props.author, [Constants.HAS_AUTHOR]);
+        } else return UNKNOWN_AUTHOR;
```
Comment (blcham) [inline src/components/comment/CommentView.jsx:22]: I saw this many times -- "else return". Why not just "return" on new line ?

File: src/components/comment/CommentView.jsx [hunk 7]
```diff
@@ -1,22 +1,50 @@
...
+        setShow(!show);
+    }
+
     return (
         <div className="comment-content">
             <div className="row">
-                <span className="col-auto comment-author">{parseJsonComment(props.author, [Constants.HAS_AUTHOR])}</span>
-                <span className="col-auto text-muted comment-timestamp">{timeAgo}</span>
+                <span className="col-auto comment-author" onMouseEnter={onMouseEventHandler} onMouseLeave={onMouseEventHandler}>
+                    {show ? getAuthor() : getSplitAuthor()}
```
Comment (blcham) [inline src/components/comment/CommentView.jsx:45]: getSplitAuthor -- strange name, i don't know what splitting means (unless i read the code ;), but you are not describing what it is but how you achieved it. What about getShortenedAutorIRI() ?

File: src/components/comment/CommentView.jsx (hunk 5 above)
Comment (LaChope) [reply src/components/comment/CommentView.jsx:14]: Yes, I just realized now... I really don't know why I did this... I wwas converting a string to Json and then converting Json to string.... I guess I was tired that day.

Comment (blcham) [reply src/components/comment/CommentView.jsx:14]: And is it resolved now ? @LaChope

File: src/components/comment/CommentView.jsx (hunk 1 above)
Comment (LaChope) [reply src/components/comment/CommentView.jsx]: @blcham This part is outdated, it is not present in the code anymore.

**Comment by blcham** [comment]
Already merged.

**Pull request closed without merging**